Fix testing on projects that use the plugin

More specifically, maven-surefire-plugin breaks and must be disabled when reactive-app-maven-plugin is used.

e.g. maven surefire (blocking Lagom team testing)

### steps

Here's the pom.xml that I am using:

```xml
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
                      http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>org.example</groupId>
    <artifactId>hello</artifactId>
    <version>1.0-SNAPSHOT</version>

    <packaging>pom</packaging>

    <modules>
        <module>hello-api</module>
        <module>hello-impl</module>
        <module>stream-api</module>
        <module>stream-impl</module>
        <module>integration-tests</module>
    </modules>

    <build>
        <plugins>
            <plugin>
                <artifactId>maven-assembly-plugin</artifactId>
                <version>2.6</version>
            </plugin>
            <plugin>
                <groupId>com.lightbend.lagom</groupId>
                <artifactId>lagom-maven-plugin</artifactId>
                <version>${lagom.version}</version>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.5.1</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                    <compilerArgs>
                        <arg>-parameters</arg>
                    </compilerArgs>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-enforcer-plugin</artifactId>
                <version>1.4.1</version>
                <executions>
                    <execution>
                        <id>enforce-maven-version</id>
                        <goals>
                            <goal>enforce</goal>
                        </goals>
                        <configuration>
                            <rules>
                                <!-- Lagom requires at least Maven version 3.2.1 -->
                                <requireMavenVersion>
                                    <version>[3.2.1,)</version>
                                </requireMavenVersion>
                            </rules>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

            <plugin>
                <groupId>com.lightbend.rp</groupId>
                <artifactId>reactive-app-maven-plugin</artifactId>
                <version>0.3.0</version>
                <configuration>
                    <mainClass>com.lightbend.rp.test.akkacluster.SimpleClusterApp</mainClass>
                </configuration>
                <executions>
                    <execution>
                        <id>build-docker</id>
                        <goals>
                            <goal>build</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>

        </plugins>
    </build>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.lightbend.lagom</groupId>
                <artifactId>lagom-maven-dependencies</artifactId>
                <version>${lagom.version}</version>
                <scope>import</scope>
                <type>pom</type>
            </dependency>
            <dependency>
                <groupId>org.projectlombok</groupId>
                <artifactId>lombok</artifactId>
                <version>1.16.10</version>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

        <scala.binary.version>2.12</scala.binary.version>
        
        <lagom.version>1.4.6</lagom.version>
    </properties>
</project>

```

### problem

```
[ERROR] DOCKER> Failed to create assembly for docker image  (with mode 'dir'): Error creating assembly archive docker: You must set at least one file.. If you include the build artifact please ensure that you have built the artifact before with 'mvn package' (should be available in the target/ dir). Please see the documentation (section "Assembly") for more information. [Error creating assembly archive docker: You must set at least one file.]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] hello .............................................. FAILURE [  2.587 s]
[INFO] hello-api .......................................... SKIPPED
[INFO] hello-impl ......................................... SKIPPED
[INFO] stream-api ......................................... SKIPPED
[INFO] stream-impl ........................................ SKIPPED
[INFO] integration-test ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.306 s
[INFO] Finished at: 2018-07-05T23:39:26-04:00
[INFO] Final Memory: 26M/256M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.lightbend.rp:reactive-app-maven-plugin:0.3.0:build (build-docker) on project hello: Unable to execute mojo: Failed to create assembly for docker image  (with mode 'dir'): Error creating assembly archive docker: You must set at least one file.. If you include the build artifact please ensure that you have built the artifact before with 'mvn package' (should be available in the target/ dir). Please see the documentation (section "Assembly") for more information. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/src/it/hello/pom.xml
+++ b/src/it/hello/pom.xml
@@ -17,7 +17,7 @@
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
-            <version>3.8.1</version>
+            <version>4.12</version>
             <scope>test</scope>
         </dependency>
     </dependencies>
@@ -68,6 +68,18 @@
                         </goals>
                     </execution>
                 </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.0</version>
+                <dependencies>
+                  <dependency>
+                    <groupId>org.apache.maven.surefire</groupId>
+                    <artifactId>surefire-junit47</artifactId>
+                    <version>2.22.0</version>
+                  </dependency>
+                </dependencies>
             </plugin>
         </plugins>
     </build>

--- a/src/it/hello/src/main/java/App.java
+++ b/src/it/hello/src/main/java/App.java
@@ -4,10 +4,12 @@ package com.lightbend.app;
  * Hello world!
  *
  */
-public class App 
+public class App
 {
+    public static String greeting = "Hello, world!";
+
     public static void main( String[] args )
     {
-        System.out.println( "Hello World!" );
+        System.out.println(greeting);
     }
 }

--- a/src/it/hello/src/test/java/AppTest.java
+++ b/src/it/hello/src/test/java/AppTest.java
@@ -1,22 +1,11 @@
 package com.lightbend.app;
 
-import junit.framework.Test;
-import junit.framework.TestCase;
-import junit.framework.TestSuite;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
 
-public class AppTest
-    extends TestCase
-{
-    public AppTest( String testName )
-    {
-        super( testName );
-    }
-    public static Test suite()
-    {
-        return new TestSuite( AppTest.class );
-    }
-    public void testApp()
-    {
-        assertTrue( true );
+public class AppTest {
+    @Test
+    public void testApp() {
+        assertEquals("Hello, world!", App.greeting);
     }
 }

--- a/src/it/lagom-endpoints/hello-impl/pom.xml
+++ b/src/it/lagom-endpoints/hello-impl/pom.xml
@@ -45,6 +45,13 @@
             <artifactId>reactive-lib-akka-cluster-bootstrap_${scala.binary.version}</artifactId>
             <version>0.8.1</version>
         </dependency>
+
+        <dependency>
+            <groupId>com.lightbend.lagom</groupId>
+            <artifactId>lagom-javadsl-testkit_${scala.binary.version}</artifactId>
+            <version>${lagom.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>
@@ -68,7 +75,7 @@
                             <goal>build</goal>
                         </goals>
                     </execution>
-                </executions> 
+                </executions>
             </plugin>
         </plugins>
     </build>

--- a/src/it/lagom-endpoints/hello-impl/src/test/java/com/lightbend/rp/test/lagomendpoints/impl/HelloServiceTest.java
+++ b/src/it/lagom-endpoints/hello-impl/src/test/java/com/lightbend/rp/test/lagomendpoints/impl/HelloServiceTest.java
@@ -1,0 +1,22 @@
+package com.lightbend.rp.test.lagomendpoints.impl;
+
+import org.junit.Test;
+
+import com.lightbend.rp.test.lagomendpoints.api.HelloService;
+
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.defaultSetup;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.withServer;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+
+public class HelloServiceTest {
+    @Test
+    public void shouldGreet() {
+        withServer(defaultSetup(), server -> {
+            HelloService service = server.client(HelloService.class);
+
+            String msg1 = service.hello("Alice").invoke().toCompletableFuture().get(5, SECONDS);
+            assertEquals("hello: Alice", msg1);
+        });
+    }
+}

--- a/src/it/lagom-endpoints/hello-impl/src/test/resources/logback.xml
+++ b/src/it/lagom-endpoints/hello-impl/src/test/resources/logback.xml
@@ -1,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <conversionRule conversionWord="coloredLevel" converterClass="com.lightbend.lagom.internal.logback.ColoredLevel" />
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%date{"HH:mm:ss.SSS"} %coloredLevel %logger - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <logger name="org.apache.cassandra" level="ERROR" />
+    <logger name="com.datastax.driver" level="WARN" />
+
+    <logger name="akka" level="WARN" />
+
+    <root level="INFO">
+        <appender-ref ref="STDOUT" />
+    </root>
+
+</configuration>

--- a/src/it/lagom-endpoints/pom.xml
+++ b/src/it/lagom-endpoints/pom.xml
@@ -75,6 +75,19 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>2.22.0</version>
+                <dependencies>
+                  <dependency>
+                    <groupId>org.apache.maven.surefire</groupId>
+                    <artifactId>surefire-junit47</artifactId>
+                    <version>2.22.0</version>
+                  </dependency>
+                </dependencies>
+            </plugin>
         </plugins>
     </build>
 

